tests/src/services/storage.service.test.ts

```diff
@@ -1,19 +1,65 @@
 import axios from 'axios';
 import jwt from 'jsonwebtoken';
 import config from '../../../src/config';
-import { getUser } from '../fixtures';
-import { getUserStorage } from '../../../src/services/storage.service';
+import { getUser, voidPromise } from '../fixtures';
+import { getUserStorage, StorageService } from '../../../src/services/storage.service';
 
 jest.mock('axios');
 
 jest.mock('jsonwebtoken', () => ({
   sign: jest.fn(() => 'mocked-jwt-token'),
 }));
 
+let storageService: StorageService;
+
+beforeEach(() => {
+  storageService = new StorageService(config, axios);
+  jest.restoreAllMocks();
+});
+
 describe('Storage service tests', () => {
-  describe('Check if the user can stack storage', () => {
-    const mockedAxios = axios as jest.Mocked<typeof axios>;
+  const mockedAxios = axios as jest.Mocked<typeof axios>;
+
+  describe('The user updates the storage', () => {
+    beforeEach(() => {
+      jest.clearAllMocks();
+    });
+
+    it('When the API request to update the user storage fails, then an error indicating so is thrown', async () => {
+      const mockedUserUuid = getUser().uuid;
+      const newStorageBytes = 100000;
+
+      mockedAxios.patch.mockImplementation(voidPromise);
+
+      await storageService.changeStorage(mockedUserUuid, newStorageBytes);
 
+      expect(jwt.sign).toHaveBeenCalledTimes(1);
+      expect(jwt.sign).toHaveBeenCalledWith({}, expect.any(String), expect.any(Object));
+      expect(mockedAxios.patch).toHaveBeenCalledTimes(1);
+      expect(mockedAxios.patch).toHaveBeenCalledWith(
+        `${config.DRIVE_NEW_GATEWAY_URL}/gateway/users/${mockedUserUuid}`,
+        { maxSpaceBytes: newStorageBytes },
+        expect.objectContaining({
+          headers: expect.objectContaining({
+            Authorization: expect.stringContaining('Bearer '),
+            'Content-Type': 'application/json',
+          }),
+        }),
+      );
+    });
+
+    it('When the updates the storage, then it should be done correctly and called with the necessary data', async () => {
```

Comment:
How is this test checking that the required parameters are being enforced?

Reply:
I have put the description of the tests backwards. Changing it.